src/main/cpp/t07_ip.cpp

```diff
@@ -29,10 +29,56 @@
 
 #include "t07_ip.h"
 #include <iostream>
-
+#include <cstring>
 
 using namespace std;
+bool isIPAddressValid(const char* pszIPAddr)
+{
+    if (!pszIPAddr) return false;
+    char IP1[100],cIP[4];
+    int len = strlen(pszIPAddr);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4513132)